packages/next/server/render.tsx

```diff
@@ -1011,7 +1011,7 @@ export async function renderToHTML(
         documentElement: (htmlProps: HtmlProps) => (
           <Document {...htmlProps} {...docProps} />
         ),
-        head: docProps.head,
+        head: docProps.head || head,
```

Comment:
Is the default head even needed if rendering outside of Next.js? It seems in the case of `getDataFromTree` the head elements won't be used and if they were meant to be used, expected head items would be missing. 

I'm gonna close this for now as this seems to need more investigation which can continue on the original mentioned thread.